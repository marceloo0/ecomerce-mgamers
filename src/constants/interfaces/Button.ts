interface Common {
  size?: 'small' | 'medium' | 'large'
}

export interface ButtonProps extends Common {
  icon?: JSX.Element
  children?: React.ReactNode
  onClick?: () => void
}

export interface WrapperProps extends Common {
  hasIcon: boolean
  hasChildren: boolean
}
