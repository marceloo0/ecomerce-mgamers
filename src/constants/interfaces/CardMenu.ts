export interface CardMenuImgStyle {
  sizeImg: 'small' | 'medium' | 'large'
}

export interface CardMenuProps {
  img: string
  price: string
  titleBackground: string
  onClick?: () => void
}
