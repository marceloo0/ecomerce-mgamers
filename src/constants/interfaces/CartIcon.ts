export interface CartIconProps {
  icon: React.ReactNode
}
