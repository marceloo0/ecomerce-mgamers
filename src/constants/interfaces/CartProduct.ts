export interface CartProductProps {
  price: string
  promotion: string
  title: string
  img: string
  titleBackground: string
  handleNextSlide: () => void
  handlePrevSlide: () => void
}

export interface CartItemProps extends CartProductProps {
  quantity: number
}

const propsCardTest = {
  price: '109',
  titleBackground: 'teclados',
  img: '/img/mouseMenu.svg'
}

export const props = {
  price: 'R$ 109,00',
  titleBackground: 'teclados',
  img: '/img/products/mouse.svg',
  promotion: 'R$ 89,00',
  title: ''
}
