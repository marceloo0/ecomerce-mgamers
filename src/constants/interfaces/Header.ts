export interface HeaderProps {
  setShowMenu?: () => void
  openModal?: () => void
  login?: boolean
}
