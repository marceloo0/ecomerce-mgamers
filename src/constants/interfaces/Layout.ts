export interface Layout {
  children: React.ReactNode
}
