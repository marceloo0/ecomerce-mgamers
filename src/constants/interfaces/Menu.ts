export const menu = [
  {
    title: 'NovidadeS',
    img: '/img/menu/news.svg',
    price: 'R$ 100,00'
  },
  {
    title: 'CadeiraS',
    img: '/img/menu/fantech.svg',
    price: 'R$ 919,90'
  },
  {
    title: 'HeadseatS',
    img: '/img/menu/headset.svg',
    price: 'R$ 319,90'
  },
  {
    title: 'MouseS',
    img: '/img/menu/mouse.svg',
    price: 'R$ 79,90'
  },
  {
    title: 'MousepadS',
    img: '/img/menu/mousepad.svg',
    price: 'R$ 69,90'
  },
  {
    title: 'TecladoS',
    img: '/img/menu/teclado.svg',
    price: 'R$ 219,33'
  }
]
