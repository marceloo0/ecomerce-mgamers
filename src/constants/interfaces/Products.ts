export interface ProductsProps {
  id: number
  img: string
  price: number
  promotion: number
  title: string
  titleBackground: string
}
export interface Stock extends ProductsProps {
  quantity: number
}
export interface ButtonProductProps extends Stock {
  handleNextSlide: () => void
  handlePrevSlide: () => void
}
export interface ProductsItemProps extends ButtonProductProps {
  item: Stock
}
export interface ItemProductProps extends Stock {
  item: Stock
}
