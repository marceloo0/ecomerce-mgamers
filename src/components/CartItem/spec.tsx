import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'
import { props } from '../../constants/interfaces/CartProduct'
import items from './mock'

import { CartItem } from '.'

describe('<CartItem />', () => {
  it('should render quantity 1', () => {
    const { container } = renderWithTheme(<CartItem {...items} quantity={1} />)

    expect(screen.getByText('R$ 109,00')).toBeInTheDocument()

    expect(screen.getByRole('img', { name: items.img })).toHaveAttribute(
      'src',
      items.img
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render quantity 2 or >', () => {
    const { container } = renderWithTheme(<CartItem {...items} quantity={3} />)

    expect(screen.getByText('R$ 109,00')).toBeInTheDocument()

    expect(screen.getByText(/3/i)).toBeInTheDocument()

    expect(screen.getByRole('img', { name: items.img })).toHaveAttribute(
      'src',
      items.img
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
