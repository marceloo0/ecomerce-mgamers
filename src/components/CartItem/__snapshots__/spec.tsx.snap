// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should render quantity 1 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 1px solid transparent;
  border-radius: 0.4rem;
  cursor: pointer;
  color: #C7ECEC;
  background-color: #201C2B;
  padding: 0.5rem;
  background-color: transparent;
}

.c6:hover {
  background-color: #332756;
}

.c6 > svg {
  color: #7FFA45;
}

.c6:hover {
  border: 1px solid #7FFA45;
  background-color: transparent;
}

.c0 {
  max-width: 1060px;
  width: 100%;
  padding: 12px;
  margin: auto;
}

.c7 {
  border-bottom: 1px solid #415151;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  padding: 12px;
  margin: auto;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c4 {
  color: #C7ECEC;
  font-size: 1.2rem;
}

.c5 {
  color: #7FFA45;
  font-size: 1.2rem;
  font-weight: 600;
}

.c3 {
  height: 9rem;
  width: 100%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt="/img/products/mouseRazer.svg"
        class="c3"
        src="/img/products/mouseRazer.svg"
      />
    </div>
    <div
      class="c2"
    >
      <h3
        class="c4"
      >
        Mouse Gamer Razer Deathadder
      </h3>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c5"
        >
          R$Â 155,90
        </span>
      </div>
      <div
        class="c2"
        style="margin-left: 24px;"
      >
        <button
          class="c6"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.5 6h-5.5v-5.5c0-0.276-0.224-0.5-0.5-0.5h-3c-0.276 0-0.5 0.224-0.5 0.5v5.5h-5.5c-0.276 0-0.5 0.224-0.5 0.5v3c0 0.276 0.224 0.5 0.5 0.5h5.5v5.5c0 0.276 0.224 0.5 0.5 0.5h3c0.276 0 0.5-0.224 0.5-0.5v-5.5h5.5c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5z"
            />
          </svg>
        </button>
        <button
          class="c6"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 6.5v3c0 0.276 0.224 0.5 0.5 0.5h15c0.276 0 0.5-0.224 0.5-0.5v-3c0-0.276-0.224-0.5-0.5-0.5h-15c-0.276 0-0.5 0.224-0.5 0.5z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <div
    class="c7"
  />
</div>
`;
